### PR TITLE
IB/GTEST: fix the usage of lmc and src_path_bits and test it in gtest.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2480/ for details.

**Comment by alinask** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1824/ for details (Mellanox internal link).

File: src/uct/ib/dc/base/dc_ep.h [hunk 1]
```diff
@@ -18,6 +18,7 @@ struct uct_dc_ep {
     uint8_t               dci;
     uint8_t               state;
     uint16_t              umr_offset;
+    uint8_t               path_bits;
```
Comment (yosefe) [inline src/uct/ib/dc/base/dc_ep.h]: need to take it from iface and not enlarge the size of dc_ep structure

Comment (alinask) [inline src/uct/ib/dc/base/dc_ep.h]: Why only for dc?

Comment (yosefe) [inline src/uct/ib/dc/base/dc_ep.h]: for rc - scale is less important

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2488/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2489/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1836/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1

**Pull request merged**